### ISSUE TITLE
Orcid.md

### ISSUE THREAD

**Issue by bieszczatkeith550** [bieszczatkeith550/KeithNumber.md#1]
curl -i -H "Accept: application/vnd.orcid+xml" -H 'Authorization: Bearer dd91868d-d29a-475e-9acb-bd3fdf2f43f4' 'https://api.sandbox.orcid.org/v2.1/0000-0002-9227-8514/works'

curl -i -H "Accept: application/vnd.orcid+xml" -H 'Authorization: Bearer dd91868d-d29a-475e-9acb-bd3fdf2f43f4' 'https://api.sandbox.orcid.org/v2.1/0000-0002-9227-8514/work/733536'

curl -i -H "Accept: application/vnd.orcid+xml" -H 'Authorization: Bearer dd91868d-d29a-475e-9acb-bd3fdf2f43f4' 'https://api.sandbox.orcid.org/v2.1/0000-0002-9227-8514/works/733535,733536'

curl -i -H 'Content-type: application/vnd.orcid+xml' -H 'Authorization: Bearer dd91868d-d29a-475e-9acb-bd3fdf2f43f4' -d '@[FILE-PATH]/work.xml' -X POST 'https://api.sandbox.orcid.org/v2.1/0000-0002-9227-8514/work'

curl -i -H 'Content-type: application/vnd.orcid+xml' -H 'Authorization: Bearer dd91868d-d29a-475e-9acb-bd3fdf2f43f4' -d '@[FILE-PATH]/works.xml' -X POST 'https://api.sandbox.orcid.org/v2.1/0000-0002-9227-8514/works'

curl -i -H 'Content-type: application/vnd.orcid+xml' -H 'Authorization: Bearer dd91868d-d29a-475e-9acb-bd3fdf2f43f4' -d '@[FILE-PATH]/work-updated.xml' -X PUT 'https://api.sandbox.orcid.org/v2.1/0000-0002-9227-8514/work/[Client ID
APP-7BD9S2NUHT6JSQOA
Client secret
05c4d454-993a-494c-adbd-7423d93d7503]'

https://sandbox.orcid.org/oauth/authorize?client_id=APP-674MCQQR985VZZQ2&response_type=code&scope=/activities/update%20/read-limited&redirect_uri=https://developers.google.com/oauthplayground

https://developers.google.com/oauthplayground/?code=WkiYjn

curl -i -L -H 'Accept: application/json' --data 'client_id=APP-674MCQQR985VZZQ2&client_secret=d08b711e-9411-788d-a474-46efd3956652&grant_type=authorization_code&code=*WkiYjn*' 'https://sandbox.orcid.org/oauth/token'

{"access_token":"f5af9f51-07e6-4332-8f1a-c0c11c1e3728","token_type":"bearer",
"refresh_token":"f725f747-3a65-49f6-a231-3e8944ce464d","expires_in":631138518,
"scope":"/activities/update /read-limited","name":"Sofia Garcia","orcid":"0000-0001-2345-6789"}

curl -i -L -H 'Accept: application/json' -d 'client_id=APP-674MCQQR985VZZQ2' -d 'client_secret=d08b711e-9411-788d-a474-46efd3956652' -d 'scope=/read-public' -d 'grant_type=client_credentials' 'https://sandbox.orcid.org/oauth/token'

<common:source-orcid>
 <common:uri>https://orcid.org/8888-8888-8888-8880</common:uri>
 <common:path>8888-8888-8888-8880</common:path>
 <common:host>orcid.org</common:host>
</common:source-orcid>

<git clone https://github.com/ORCID/orcid-angular>
<cd orcid-angular
yarn>
<yarn start-local>
<<yarn build:i18n>>
<yarn start:en ## Runs the application in using the english properties
yarn start:fr ## Runs the application in using the french properties
yarn start:ar ## Runs the application in using the arabic properties
yarn start:es ## Runs the application in using the spanish properties
yarn start:ca ## ... and so on
yarn start:cs
yarn start:it
yarn start:ja
yarn start:ko
yarn start:pt
yarn start:ru
yarn start:uk
yarn start:zh_CN
yarn start:zh_TW
yarn start:lr ## These last fourth configurations are used only for testing
yarn start:rl
yarn start:xx
yarn start:source>

ng generate component component-name

ng generate --help

<ng build orcid-tokens>

<cd dist/orcid-tokens>

<npm publish>

ng test
Running end-to-end tests

For end-to-end (e2e) testing, run:

ng e2e

# dependencies docker build

# match version from .tool-versions
FROM maven:3.6.3-jdk-11 AS maven

ARG tag_numeric

WORKDIR /build

# copy only poms for max cachability of just dependency downloads
COPY pom.xml .
COPY orcid-core/pom.xml orcid-core/pom.xml
COPY orcid-persistence/pom.xml orcid-persistence/pom.xml
COPY orcid-utils/pom.xml orcid-utils/pom.xml
COPY orcid-test/pom.xml orcid-test/pom.xml
COPY orcid-api-common/pom.xml orcid-api-common/pom.xml
COPY orcid-scheduler-web/pom.xml orcid-scheduler-web/pom.xml
COPY orcid-api-web/pom.xml orcid-api-web/pom.xml
COPY orcid-message-listener/pom.xml orcid-message-listener/pom.xml
COPY orcid-core/pom.xml orcid-core/pom.xml
COPY orcid-web/pom.xml orcid-web/pom.xml
COPY orcid-internal-api/pom.xml orcid-internal-api/pom.xml
COPY orcid-pub-web/pom.xml orcid-pub-web/pom.xml
COPY orcid-activemq/pom.xml orcid-activemq/pom.xml

# FIXME: these dont seem required?
#COPY orcid-web-frontend/pom.xml orcid-web-frontend/pom.xml
#COPY orcid-activities-indexer/pom.xml orcid-activities-indexer/pom.xml
#COPY orcid-nodejs/pom.xml orcid-nodejs/pom.xml

# download maven dependencies and ignore that some components will fail
RUN mvn -T 1C --batch-mode dependency:resolve --fail-never -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn

# install orcid-parent into our local maven repo because the builds depend a version tagged release
RUN mvn -T 1C --batch-mode --non-recursive clean install -DskipTests \
-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn

# install orcid-utils into our local maven repo because the builds depend a version tagged release
COPY orcid-utils/src orcid-utils/src
RUN mvn -T 1C --batch-mode --projects orcid-utils clean install -DskipTests \
-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn

# install orcid-test into our local maven repo because orcid-persistence depends on it
COPY orcid-test/src orcid-test/src
RUN mvn -T 1C --batch-mode --projects orcid-test clean install -DskipTests \
-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn

# install orcid-persistence into our local maven repo because orcid-core depends on it
COPY orcid-persistence/src orcid-persistence/src
RUN mvn -T 1C --batch-mode --projects orcid-persistence clean install -DskipTests \
-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn

# install orcid-core into our local maven repo because the builds depend a version tagged release
COPY orcid-core/src orcid-core/src
RUN mvn -T 1C --batch-mode --projects orcid-core clean install -DskipTests \
-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn

# install orcid-api-common into our local maven repo because orcid-web deploy depends a version tagged release
COPY orcid-api-common/src orcid-api-common/src
RUN mvn -T 1C --batch-mode --projects orcid-api-common clean install -DskipTests \
-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn

##################################################################################
COPY orcid-web/src orcid-web/src
RUN mvn -T 1C --batch-mode -DgenerateBackupPoms=false \
--projects orcid-web -am package -DskipTests \
-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn

# For Java 11 and Tomcat 9
FROM tomcat:9.0.93-jdk11-temurin-jammy

# Focal has no j2cli support
# FROM tomcat:9.0.91-jdk11-temurin-focal

# copy jar file from build
COPY --from=maven /build/*/target/*.war /usr/local/tomcat/webapps/orcid-web.war

RUN mkdir -p /usr/local/tomcat/newrelic
COPY newrelic.yml /usr/local/tomcat/newrelic/newrelic.yml
RUN curl -L -s https://download.newrelic.com/newrelic/java-agent/newrelic-agent/8.13.0/newrelic-agent-8.13.0.jar -o /usr/local/tomcat/newrelic/newrelic.jar

RUN apt-get update
RUN apt-get install -y j2cli

# add orcid ca to allow Java application to trust other containers
ADD certs/cacerts /opt/java/openjdk/lib/security/cacerts

# add orcid ca to system to allow curl healthchecks to work
ADD certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt

COPY entrypoint.sh .
RUN chmod +x ./entrypoint.sh

COPY orcid-web/*.j2 .

COPY orcid-web/log4j2.xml .

ENTRYPOINT ./entrypoint.sh

cd user-service-2      ; ./mvnw   # http://localhost:9000
cd member-service-2    ; ./mvnw   # http://localhost:9010
cd assertion-service-2 ; ./mvnw   # http://localhost:9020

cd ui-2
npm install
npm run start          # http://localhost:4200

<<./docker-build.sh>>

![image](https://github.com/user-attachments/assets/c463b6da-79cf-4da4-91fe-bb16b854bc80)
